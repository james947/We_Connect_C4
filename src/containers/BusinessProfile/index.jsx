import React, { Component } from 'react';
import axios from 'axios';
import BusinessesProfile from '../../components/BusinessProfile';
import swal from 'sweetalert';

class ProfilePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      reviews: []
    };
  }

  componentDidMount() {
    axios.get(`https://weconnect-v3.herokuapp.com/api/v1/businesses/${this.props.match.params['id']}`)
      .then(response => this.setState({ data: response.data }))

      .catch(err=>{
        swal(err.response.data.message)
        window.location.assign('/businesses')     
  
      })
      
      this.fetchReviews();

  }

  fetchReviews = () => {
    axios.get(`https://weconnect-v3.herokuapp.com/api/v1/businesses/${this.props.match.params['id']}/reviews`)
      .then(response => this.setState({ reviews: response.data }))
  }
  
  render() {
    return (
      <BusinessesProfile fetchReviews={ this.fetchReviews } state={this.state}/>


    );
  }
}

export default ProfilePage;
