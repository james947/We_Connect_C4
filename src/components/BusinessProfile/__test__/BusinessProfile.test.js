import { shallow } from 'enzyme'
import React from 'react';
import  BusinessProfile  from '../';

describe("<BusinessProfile/>", () => {
    it('renders correctly', () => {
        const wrapper = shallow(<BusinessProfile state={{data: {}, reviews: {}}}/>);
        expect(wrapper).toMatchSnapshot();
    })
})