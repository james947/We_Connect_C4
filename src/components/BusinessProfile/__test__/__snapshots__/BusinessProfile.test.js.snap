// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessProfile/> renders correctly 1`] = `
<div>
  <Connect(NavBar) />
  <div
    className="row content SideNav container"
  >
    <div
      className="col-md-3 jumbotron"
    >
      <Connect(SideNav)
        state={Object {}}
      />
    </div>
     
    <div
      className="col-md-9"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Profile
        </div>
        <div
          className="card-body ProfileCard"
        >
          <h5>
            Business Name
          </h5>
          <small>
            <p />
          </small>
          <h5>
            Description
          </h5>
          <small>
            <p />
          </small>
          <h5>
            Category
          </h5>
          <small>
            <p />
          </small>
          <h5>
            Location
          </h5>
          <small>
            <p />
          </small>
        </div>
      </div>
    </div>
     
    <div
      className="col-md-3"
    />
    <div
      className="col-md-9 Reviews"
    >
      <Connect(Component)
        business={Object {}}
        state={Object {}}
      />
    </div>
     
  </div>
  <Footer />
</div>
`;
